src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -207,8 +211,15 @@ public BuildData(Run<?, ?> build, Date currentTime, TaskListener listener) {
     }
   }
 
+  // ISO 8601 date format
+  public static DateFormat isoDateFormatter() {
+    return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```

Comment:
This is a potential performance bottleneck. Every time you create a new instance which means that the formatter needs to parse the string. Having just one instance as before which is reused should be faster

Reply:
SimpleDateFormat is not thread safe, as pointed out by Findbugs :)

If we care about performance overhead the easiest replacement is https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/time/FastDateFormat.html

Reply:
.. or this from JDK https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

Reply:
I don't see DateTimeFormatter available on Java 7. Do you want the apache one or keep current approach. Btw if we use the apache commons formatter we will still change API compatibility as it is only a `Format`, not `DateFormat`.

Reply:
Interesting comparison http://optimizationtricks.blogspot.bg/2015/01/datetimeformat-comparison-java-vs-joda.html , please tell me.

Reply:
re. Java 8: right I keep forgetting that it's still building on 7

re. format only: AFAIK we only use it to format, anyhow since 3.2 it supports parsing too

I'm fine with either Joda or FastDateFormat